Proposals/0002-date-interval-differences.md

```diff
@@ -0,0 +1,66 @@
+
+Should a developer find themselves needing to compute these values, the proposed additions should feel intuitive because they bear a family resemblance to existing `Date` APIs (e.g.: `timeIntervalSince(_:)`). These additions build on and complement existing intersection and comparision APIs. By easily indicating if two intervals intersect for a moment at opposing ends, developers can more easily determine special cases of intersection. By indicating if one interval is exclusively relative to another in the past or the future, the comparison offering is enriched. The names for these computations are expressive, promoting clarity and readability, especially when developers may be tempted to settle for terse expressions. With the quality of these APIs assured; the convenience saving developers the cognitive load implementing, naming, testing and maintaining these computations and; the presence of these small additions that congeal with intuitions about Foundation types; I would propose these additions are delightful.
+
+## Detailed design
+
+Consider two date intervals `A` and `B` demarcating ascending moments on this first timeline:
+```
+1. |--A--| <-- time interval --> |--B--|
+```
+For the `timeIntervalSince(_:)` APIs, the time interval of `B` since `A` will be zero or positive because `B` demarcates a range of moments later than the range of moments demarcated by `A`. The time interval of `A` since `B` will be zero or negative because the range of moments the range of moments demarcated by `A` are earlier than those demarcated by `B`.
+
+For the `dateIntervalSince(_:)` APIs, the date interval of `B` since `A` will be the same as the date interval of `A` since `B` because `DateInterval` is designed to represent positive intervals only. The time between the earliest end and latest start becomes the duration and the earliest end becomes the start of the resulting `DateInterval`.
+
+Next, consider two date intervals `C` and `D` demarcating moments on a second timeline:
+```
+2. |--C--|--D--|
+```
+The time or date intervals since `C` and `D` and vice-versa are zero because `C` ends when `D` begins. With this, developers can test that two date intervals form a seamless, continuous range of moments.
+
+Finally, consider two thoroughly intersecting date intervals demarcating these moments on a third timeline:
+```
+3. |--|—-—-|--|
+```
+In this case, the time or date intervals since either date interval does not exist, so the APIs shall return `nil`.
+
+## Source compatibility
+
+The proposed changes are additive and no significant impact on existing code is expected.
+
+## Implications on adoption
+
+These additions can be freely adopted and un-adopted in source code with no deployment constraints and without affecting source compatibility.
+
+## Future directions
+
+`DateInterval` could gain more API inspired by operations in set theory. I would expect these proposed additions can be subsumed and compatible with such a direction.
```

Comment:
It'd be useful if you can give a few concrete examples of these. But if they're not clearly related to this proposal I'd recommend leaving them out from this proposal